source/robot_model/src/Model.cpp

```diff
@@ -346,4 +346,43 @@ void Model::print_qp_problem() {
     std::cout << this->upper_bound_constraints_(i) << std::endl;
   }
 }
+
+bool Model::in_range(const state_representation::JointPositions& joint_positions) {
+  return this->in_range(joint_positions.get_positions(), this->robot_model_.lowerPositionLimit,
+                                                         this->robot_model_.upperPositionLimit);
+}
+
+bool Model::in_range(const state_representation::JointVelocities& joint_velocities) {
+  return this->in_range(joint_velocities.get_velocities(), -this->robot_model_.velocityLimit,
+                                                            this->robot_model_.velocityLimit);
+}
+
+bool Model::in_range(const state_representation::JointTorques& joint_torques) {
+  return this->in_range(joint_torques.get_torques(), -this->robot_model_.effortLimit,
+                                                      this->robot_model_.effortLimit);
+}
+
+bool Model::in_range(const state_representation::JointState& joint_states) {
```

Comment:
Is the model also giving you acceleration limits? If yes you might want to add it as well

Reply:
Unfortunately not. The urdf provides only angles, torque and velocity.
Anyway the acceleration depends on the torque and the mass, so it will be limited by the effortLimit.

```
<joint name="${arm_id}_joint3" type="revolute">
     ...
      <limit effort="87" lower="-2.8973" upper="2.8973" velocity="2.1750"/>
      ...
    </joint>
```